frontend/src/components/document/DocumentView/StyledText.jsx

```diff
@@ -1,91 +1,107 @@
 import { getCommentThreadsOnTextNode, ifMaybeCommentOnTextNode } from "@/util/editorCommentUtils";
 import { getRedactionsOnTextNode, getMarkFromLeaf } from "@/util/editorRedactionUtils";
-import CommentedText from "@/components/document/Comments/CommentedText";
-import SuggestedText from "@/components/document/Redactions/SuggestedText";
-import RejectedText from "@/components/document/Redactions/RejectedText";
-import RedactionPopover from "@/components/document/Redactions/RedactionPopover";
 import { changeRedaction, SUGGESTION_PREFIX, ACCEPTED_PREFIX, REJECTED_PREFIX } from "@/util/editorRedactionUtils";
 
 import { useSlate } from "slate-react"
+import { useState, useEffect } from "react";
 
-import AcceptedText from "@/components/document/Redactions/AcceptedText";
+import RedactionPopover from "@/components/document/Redactions/RedactionPopover";
 import CommentPopover from "@/components/document/Comments/CommentPopover";
+import HighlightedText from "@/components/document/Redactions/HighlightedText";
+import AcceptedText from "@/components/document/Redactions/AcceptedText"
+import RejectedText from "@/components/document/Redactions/RejectedText"
+import CommentedText from "@/components/document/Comments/CommentedText"
 
 // for table view, pass in false for  isPopoverDisabled to disable popovers
 export default function StyledText({ attributes, children, leaf, isPopoverDisabled }) {
 
   const mark = getMarkFromLeaf(leaf)
   const editor = useSlate();
+  const [color, setColor] = useState('transparent'); 
 
+  //check what marks are on the textnode
   const commentThreads = getCommentThreadsOnTextNode(leaf);
+  
   const maybeComment= ifMaybeCommentOnTextNode(leaf);
+  const isSuggestion = getRedactionsOnTextNode(leaf, SUGGESTION_PREFIX).size > 0;
+  const isRejected = getRedactionsOnTextNode(leaf, REJECTED_PREFIX).size > 0;
+  const isAccepted = getRedactionsOnTextNode(leaf, ACCEPTED_PREFIX).size > 0;
+  const ifComment = commentThreads.size > 0;
 
-  if (maybeComment) {
-    return (
-      <CommentedText
-        {...attributes}
-        commentThreads={commentThreads}
-        textnode={leaf}
-      >
-        <CommentPopover
-          text={<span>{children}</span>}
-          ifOpen={true}
-        />
-      </CommentedText>
-    );
-  }
+  const isRedaction = isSuggestion || isAccepted || isRejected
 
-  if (commentThreads.size > 0) {
-    return (
-      <CommentedText
-        {...attributes}
-        commentThreads={commentThreads}
-        textnode={leaf}
-      >
-        {children}
-      </CommentedText>
-    );
-  }
+  useEffect(() => {
+    let newColor = 'transparent';
+    if (isSuggestion) {
+      newColor = (maybeComment || ifComment) ? 'suggestion-and-comment' : 'suggested-redaction';
+    } else if (isRejected) {
```

Comment:
maybe combine this one and isAccepted since its the same logic for both, like `if (isRejected || isAccepted)`

Reply:
and a written comment describing the states wouldn;t hurt either